src/device/device-management.ts

```diff
@@ -70,28 +103,67 @@ export async function createCameraTrack(
  * @returns A LocalTrack object or an error.
  */
 export async function createMicrophoneTrack(
-  constraints?: AudioDeviceConstraints
+  constraints?: MicrophoneConstraints
 ): Promise<LocalMicrophoneTrack> {
   let stream: MediaStream;
   try {
-    stream = await media.getUserMedia({ audio: { ...constraints } });
+    stream = await media.getUserMedia({ audio: generateAudioConstraints(constraints) });
   } catch (error) {
-    throw new WcmeError(
+    throw new WebrtcError(
       ErrorTypes.CREATE_MICROPHONE_TRACK_FAILED,
       `Failed to create microphone track ${error}`
     );
   }
-  return new LocalMicrophoneTrack(stream);
+  // See if we can just pass the track and not streams
+  return new LocalMicrophoneTrack(stream.getAudioTracks()[0]);
 }
 
 /**
  * Creates a display video track.
  *
+ * @param constraints - Display constraints for screen sharing.
  * @returns A Promise that resolves to a LocalDisplayTrack.
  */
-export async function createDisplayTrack(): Promise<LocalDisplayTrack> {
-  const stream = await media.getDisplayMedia({ video: true });
-  return new LocalDisplayTrack(stream);
+export async function createDisplayTrack(
+  constraints?: VideoConstraints
+): Promise<LocalDisplayTrack> {
+  const stream = await media.getDisplayMedia({ video: generateVideoConstraints(constraints) });
```

Comment:
we probably need a different type for display track constraints argument (and different method for generating them from the supplied arg), because browsers have some specific things for screen sharing, for example see here: https://developer.chrome.com/docs/web-platform/screen-sharing-controls/ for recent additions in Chrome to getDisplayMedia constraints - we should expose them to the clients

Reply:
i agree can do that in a separate PR

Reply:
to confirm will add to a separate PR

Reply:
we should have a jira for this under your [epic](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-375070)

Reply:
created display track here